app/controllers/private_pages_controller.rb

```diff
@@ -1,8 +1,55 @@
 class PrivatePagesController < ApplicationController
   before_action :authenticate_user!
+  helper_method :ban_group_status
 
   def show
     @page = params[:id]
     raise "Invalid page: #{@page.inspect}" unless File.exist?(Rails.root.join("app/views/private_pages/_#{@page}.html.slim"))
   end
+
+  def bans
+    @channels = Ban.group(:channel).order(:channel).pluck(:channel, 'count(distinct bangroup_id)')
+    render :channels
+  end
+
+  def channel_bans
+    @channel = '#' + params[:channel]
+    @bans = Ban.where(channel: @channel)
+               .order(:bangroup_id)
+               .reverse_order
+               .order(:banned_at, :id)
+               .all
```

Comment:
Looks like it's not. It just marks the border between ActiveRecord and Ruby Array calls. I'll look at removing that.